tests/robustness/multiclient_test/storagestats/storage_stats.go

```diff
@@ -0,0 +1,117 @@
+//go:build darwin || (linux && amd64)
+// +build darwin linux,amd64
+
+// Package storagestats contains logging mechanism
+// log disk space consumed by directories created by
+// robustness test framework before and after the test run.
+package storagestats
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/kopia/kopia/tests/robustness/multiclient_test/framework"
+)
+
+var logFilePath string
+
+// DirDetails represents details about a directory,
+// path, and size.
+type DirDetails struct {
+	DirPath string `json:"dirPath"`
+	DirSize int64  `json:"dirSize"`
+}
+
+// LogStorageStats logs disk space usage of provided dir paths.
+func LogStorageStats(ctx context.Context, dirs []string) error {
+	dd := collectDirDetails(dirs)
+
+	// write dir details into a JSON file
+	jsonData, err := json.Marshal(dd)
+	if err != nil {
+		log.Printf("Error marshaling to JSON %s", err)
+		return err
```

Comment:
we just want to wrap this error and return it to the caller side. The caller side will print it for us.

```
return fmt.Errorf("error marshaling to JSON: %w", err)
```

On the caller side the `log.Printf` statement will be executed, as:

```
err := storagestats.LogStorageStats(ctx, dirs)
if err != nil {
    log.Printf("Error collecting the logs: %v\n", err)
}
```
and it will print 
```
2009/11/10 23:00:00 Error collecting the logs: error marshaling to JSON: some marshaling error..
```

Reply:
done